File: androidutilslibrary/src/main/java/me/com/androidutilslibrary/NetworkUtils/NetworkUtils.java
package me.com.androidutilslibrary.NetworkUtils;

import android.content.Context;
import android.content.DialogInterface;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.support.v7.app.AlertDialog;

public class NetworkUtils {
    /**
     *
     * @param context
     * @return
     */
    public static boolean isOnline(Context context) {
        if (context != null) {
            ConnectivityManager cm =
                    (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            if (cm != null) {
                NetworkInfo netInfo = cm.getActiveNetworkInfo();
                return netInfo != null && netInfo.isConnectedOrConnecting();
            }
            return false;
        }
        return false;
    }

    /**
     *
     * @param context
     * @param title
     * @param message
     * @param positiveButtonStringValue
     * @return
     */
    public static AlertDialog.Builder buildNoNetworkDialog(Context context, String title, String message, String positiveButtonStringValue) {

        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle(title);
        builder.setMessage(message);

        builder.setPositiveButton(positiveButtonStringValue, new DialogInterface.OnClickListener() {

            @Override
            public void onClick(DialogInterface dialog, int which) {

                dialog.dismiss();
            }
        });

        return builder;
    }
}

File: androidutilslibrary/src/main/java/me/com/androidutilslibrary/networkutils/NetworkUtils.kt
package me.com.androidutilslibrary.networkutils

import android.content.Context
import android.net.ConnectivityManager

object NetworkUtils {
    /**
     *
     * @param context context
     * @return true if mobile has internet connection otherwise return false
     */
    @JvmStatic
    fun isOnline(context: Context?): Boolean {
        context?.let {
            val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return try {
                val netInfo = cm.activeNetworkInfo
                netInfo != null && netInfo.isConnectedOrConnecting

            } catch (exception: Exception) {
                false
            }
        }
        return false
    }
}